code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/VolumeDaoImpl.java

```diff
@@ -53,4 +60,45 @@ public void createVolumeInStoragePool(Map<String, Object> volumeData, StoragePoo
         vspm.put("storagePoolId", storagePool.getId());
         resourceDao.createAndSchedule(VolumeStoragePoolMap.class, vspm);
     }
+
+    @Override
+    public Volume findSharedVolume(long accountId, Long storagePoolId, String volumeName) {
+        List<VolumeRecord> volumes = null;
+        if (storagePoolId == null) {
+            volumes = create()
+            .select(VOLUME.fields())
+            .from(VOLUME)
+            .join(VOLUME_STORAGE_POOL_MAP)
+                .on(VOLUME_STORAGE_POOL_MAP.VOLUME_ID.eq(VOLUME.ID))
+            .join(STORAGE_POOL)
+                .on(VOLUME_STORAGE_POOL_MAP.STORAGE_POOL_ID.eq(STORAGE_POOL.ID)
+                .and(STORAGE_POOL.KIND.notIn(LOCAL_POOL_KINDS)))
+            .where(VOLUME.NAME.eq(volumeName)
```

Comment:
@cjellick sorry, Craig, my bad. My thoughts clearly ran somewhere while I was typing that comment. We should check for Storage_pool.REMOVED field not being null, not that record being null.